### ISSUE TITLE
How is lr calculated?

### ISSUE THREAD

**Issue by ebilal** [OpenNMT/OpenNMT-py#1666]
When training with -decay_method rsqrt -warmup_steps 0 -learning_rate 1e-4 the first learning rate displayed is 1e-5. Shouldn't it be 1e-4?

-E

**Comment by francoishernandez** [comment]
With `-decay_method rsqrt -warmup_steps 0 -learning_rate 1e-4`:
`[2019-12-06 12:12:07,837 INFO] Step  1/   50; acc:   0.00; ppl: 6223.94; xent: 8.74; lr: 0.00007;  36/ 37 tok/s;     48 sec`
I think the only 'issue' is that the logged LR is the one for `step+1` (1e-4 * 1/sqrt(2)), but otherwise it seems okay.
